microsoft-edge/extensions-chromium/developer-guide/devtools-extension.md

```diff
@@ -0,0 +1,430 @@
+
+    ```html
+    <button id="sayHello">Say Hello to The Inspected Page!</button>
+    <h2><span id="youClickedOn"></span></h2>
+    ```
+
+1. In the `devtools.js` file, use the `chrome.runtime.connect` method to create a connection to the background service worker, then send the inspected window `tabId` to the service worker with the `backgroundPageConnection.postMessage` method. Finally, add event listeners to the `sayHello` button and `youClickedOn` label defined in the  `panel.html` file.
+
+    ```javascript
+    let youClickedOn; 
+    chrome.devtools.panels.create("Sample Panel", "icon.png", "panel.html", panel => {
+        // code invoked on panel creation
+        panel.onShown.addListener( (extPanelWindow) => {
+            let sayHello = extPanelWindow.document.querySelector('#sayHello');
+            youClickedOn = extPanelWindow.document.querySelector('#youClickedOn');
+            sayHello.addEventListener("click", () => {
+                // show a greeting alert in the inspected page
+                chrome.devtools.inspectedWindow.eval('alert("Hello from the DevTools Extension");');
+            });             
+        });
+    });
+
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        // Messages from content scripts should have sender.tab set
+        if (sender.tab && request.click == true) {
+            console.log('I am here!');
+            if (youClickedOn) {
+                youClickedOn.innerHTML = `You clicked on position (${request.xPosition}, ${request.yPosition}) in the inspected page.`;
+            }
+            sendResponse({
+                xPosition: request.xPosition,
+                yPosition: request.yPosition
+            });
+        }
+    });
+
+    // Create a connection to the background service worker
+    const backgroundPageConnection = chrome.runtime.connect({
+        name: "devtools-page"
+    });
+
+    // Relay the tab ID to the background service worker
+    backgroundPageConnection.postMessage({
+        name: 'init',
+        tabId: chrome.devtools.inspectedWindow.tabId
+    });
+    ```
+
+    When the user clicks the `sayHello` button, the DevTools extension will execute a code snippet of `alert("Hello from the DevTools Extension");` in the inspected window by invoking the `eval()` method of the inspected window `chrome.devtools.inspectedWindow`.
+
+    When the user clicks anywhere in the inspected window, the DevTools extension will get a message, from the background service worker, with `request.click == true` and the mouse position information.
```

Comment:
```suggestion
    When the user clicks anywhere in the inspected window, the DevTools extension will receive a message, from the background service worker, with `request.click == true` and the mouse position information.
```
tech review this suggestion, how active is the getting on the part of this actor